building/setup_development_environment.adoc

```diff
@@ -0,0 +1,196 @@
+should see in your ~/.m2/settings.xml
+
+[source,xml]
+----
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+  <profiles>
+    <profile>
+    <id>jboss-default</id>
+    <repositories>
+    <!-- To resolve parent artifact -->
+    <repository>
+        <id>jboss-public-repository-group</id>
+        <name>JBoss Public Repository Group</name>
+        <url>http://repository.jboss.org/nexus/content/groups/public/</url>
+      </repository>
+      <repository>
+        <id>jboss-snapshots-repository</id>
+        <name>JBoss Snapshots Repository</name>
+        <url>https://repository.jboss.org/nexus/content/repositories/snapshots/</url>   
+      </repository>
+    </repositories>
+    <pluginRepositories>
+      <!-- To resolve parent artifact -->
+      <pluginRepository>
+        <id>jboss-public-repository-group</id>
+        <name>JBoss Public Repository Group</name>
+        <url>http://repository.jboss.org/nexus/content/groups/public/</url>   
+      </pluginRepository>
+      <pluginRepository>
+        <id>jboss-snapshots-repository</id>
+        <name>JBoss Snapshots Repository</name>
+        <url>https://repository.jboss.org/nexus/content/repositories/snapshots/</url>   
+      </pluginRepository>
+    </pluginRepositories>
+    </profile>
+  </profiles>
+  <activeProfiles>
+    <activeProfile>jboss-default</activeProfile>
+  ...
+  </activeProfiles>
+</settings>
+----
+
+
+Maven & Java Memory Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To configure the amount of memory used by Maven, you can define MVN_OPTS
+as follows, either in the mvn / mvn.bat script you use to run Maven, or
+set as global environment variables. Here's how to do so for
+http://forums.fedoraforum.org/showthread.php?t=262465[Fedora],
+https://help.ubuntu.com/community/EnvironmentVariables[Ubuntu],
+http://forums.techarena.in/windows-xp-support/1152405.htm[Windows],
+http://www.digitaledgesw.com/node/31[OSX].
+
+[source,bash]
+----
+set MAVEN_OPTS=-Xms512m -Xmx1024m -XX:PermSize=128m -XX:MaxPermSize=256m
+----
+
+
+==== m2e
+
+m2e (maven integation for Eclipse) is also required to get the most out of your development experience with JBoss Tools. JBoss Tools uses Maven, and a maven plugin 
+Tycho, to build our tools. To see if your chosen  distribution comes with m2e, you can check http://www.eclipse.org/downloads/compare.php?release=luna[here]. 
+If your chosen Eclipse installation does not come with m2e, you'll want to follow the above instructions to install m2e, using the update site
+url found at m2e's http://download.eclipse.org/technology/m2e/releases/[download page]. 
+
+
+Once m2e is installed, we recommend that you also install *m2e-tycho* and *m2e-EGit* connector from `Window -> Preferences -> Maven -> Discovery -> Open Catalog`.
```

Comment:
The doc in this PR about importing into eclipse shows screenshots of it being suggested to install during import.

One of these section must be wrong then. I suggest it is this one since import is supposed to do this.